subprojects/docs/src/docs/userguide/directory_layout.adoc

```diff
@@ -0,0 +1,104 @@
+<5> Global <<init_scripts.adoc#init_scripts, initialization scripts>>
+<6> Distributions downloaded by the <<gradle_wrapper.adoc#gradle_wrapper,Gradle Wrapper>>
+<7> Global <<build_environment.adoc#sec:gradle_configuration_properties,Gradle configuration properties>>
+
+[[directory_layout:gradle_user_home_directory:cache_cleanup]]
+=== Global cache cleanup
+
+Since Gradle 4.10, the Gradle user home directory is being cleaned up automatically. The cleanup is performed in the background when the Gradle daemon is stopped or shuts down. If using `--no-daemon`, it runs in the foreground after the build session with a visual progress indicator.
+
+The following cleanup strategies are applied periodically (at most every 24 hours):
+
+- Version-specific caches in `caches/<gradle-version>/` are checked for whether they are still in use. If not, directories for release versions are deleted after 30 days of inactivity, snapshot versions after 7 days of inactivity.
+- Gradle distributions in `wrapper/dists/` are checked for whether they are still in use, i.e. whether there's a corresponding version-specific cache directory. Unused distributions are deleted.
+- Shared caches in `caches/` (e.g. `jars-*`) are checked for whether they are still in use. If there's no Gradle version that still uses them, they are deleted.
+- Files in shared caches used by the current Gradle version in `caches/` (e.g. `jars-3` or `modules-2`) are checked for when they were last accessed. Depending on whether the file can be recreated locally or would have to be downloaded from a remote repository again, it will be deleted after 7 or 30 days of not being accessed, respectively.
+
+
+[[directory_layout:project_root_directory]]
+== Project root directory
+
+The project root directory has roughly the following structure:
+
+[listing,subs=+macros]
+----
+.
+├── .gradle // <1>
+│   ├── 4.8 // <2>
+│   ├── 4.9 // <2>
+│   └── ⋮
+├── build // <3>
+├── gradle
+│   └── wrapper  // <4>
+├── build.gradle pass:quotes[_or_] build.gradle.kts // <5>
+├── gradle.properties // <6>
+├── gradlew // <7>
+├── gradlew.bat // <7>
+└── settings.gradle pass:quotes[_or_] settings.gradle.kts // <8>
+----
+<1> Project-specific cache directory generated by Gradle
+<2> Version-specific caches (e.g. to support incremental builds)
+<3> The build directory of this project into which Gradle generates all build artifacts.
+<4> Contains the JAR file and configuration of the <<gradle_wrapper.adoc#gradle_wrapper,Gradle Wrapper>>
+<5> The project's Gradle build script
+<6> Project-specific <<build_environment.adoc#sec:gradle_configuration_properties,Gradle configuration properties>>
+<7> Scripts for executing builds using the <<gradle_wrapper.adoc#gradle_wrapper,Gradle Wrapper>>
+<8> The project's <<build_lifecycle.adoc#sec:settings_file, settings file>>
+
+[[directory_layout:project_root_directory:cache_cleanup]]
+=== Project cache cleanup
```

Comment:
Same comment about an awkward transition applies here too.  I would prefer fuller explanations for each of the directories. They don't have to be in-depth, but it seems odd explaining the directory structure without more information on what those directories do and how Gradle uses them.